core/capabilities/ccip/ccipsolana/commitcodec_test.go

```diff
@@ -0,0 +1,133 @@
+package ccipsolana
+
+import (
+	"math/big"
+	"math/rand"
+	"testing"
+
+	solanago "github.com/gagliardetto/solana-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+)
+
+var randomCommitReport = func() cciptypes.CommitPluginReport {
+	pubkey, err := solanago.NewRandomPrivateKey()
+	if err != nil {
+		panic(err)
+	}
+
+	return cciptypes.CommitPluginReport{
+		MerkleRoots: []cciptypes.MerkleRootChain{
+			{
+				OnRampAddress: cciptypes.UnknownAddress(pubkey.PublicKey().String()),
+				ChainSel:      cciptypes.ChainSelector(rand.Uint64()),
+				SeqNumsRange: cciptypes.NewSeqNumRange(
+					cciptypes.SeqNum(rand.Uint64()),
+					cciptypes.SeqNum(rand.Uint64()),
+				),
+				MerkleRoot: utils.RandomBytes32(),
+			},
+		},
+		PriceUpdates: cciptypes.PriceUpdates{
+			TokenPriceUpdates: []cciptypes.TokenPrice{
+				{
+					TokenID: "C8WSPj3yyus1YN3yNB6YA5zStYtbjQWtpmKadmvyUXq8",
+					Price:   cciptypes.NewBigInt(big.NewInt(rand.Int63())),
+				},
+			},
+			GasPriceUpdates: []cciptypes.GasPriceChain{
+				{GasPrice: cciptypes.NewBigInt(big.NewInt(rand.Int63())), ChainSel: cciptypes.ChainSelector(rand.Uint64())},
+				{GasPrice: cciptypes.NewBigInt(big.NewInt(rand.Int63())), ChainSel: cciptypes.ChainSelector(rand.Uint64())},
+				{GasPrice: cciptypes.NewBigInt(big.NewInt(rand.Int63())), ChainSel: cciptypes.ChainSelector(rand.Uint64())},
+			},
+		},
+	}
+}
+
+func TestCommitPluginCodecV1(t *testing.T) {
+	testCases := []struct {
+		name   string
+		report func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport
+		expErr bool
+	}{
+		{
+			name: "base report",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				return report
+			},
+		},
+		{
+			name: "empty token address",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				report.PriceUpdates.TokenPriceUpdates[0].TokenID = ""
+				return report
+			},
+			expErr: true,
+		},
+		{
+			name: "empty merkle root",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				report.MerkleRoots[0].MerkleRoot = cciptypes.Bytes32{}
+				return report
+			},
+		},
+		{
+			name: "zero token price",
+			report: func(report cciptypes.CommitPluginReport) cciptypes.CommitPluginReport {
+				report.PriceUpdates.TokenPriceUpdates[0].Price = cciptypes.NewBigInt(big.NewInt(0))
+				return report
+			},
+		},
+		{
+			name: "zero gas price",
```

Comment:
as per above comment, add another case for empty gas price, which should expect error